packages/wallet-sdk/src/sign/scw/utils/createSmartAccount.ts

```diff
@@ -0,0 +1,830 @@
+import type { Address, TypedData } from 'abitype';
+import * as Signature from 'ox/Signature';
+import type * as WebAuthnP256 from 'ox/WebAuthnP256';
+import {
+  Assign,
+  BaseError,
+  type Client,
+  decodeFunctionData,
+  encodeAbiParameters,
+  encodeFunctionData,
+  encodePacked,
+  type Hash,
+  hashMessage,
+  hashTypedData,
+  type Hex,
+  type LocalAccount,
+  type OneOf,
+  parseSignature,
+  type Prettify,
+  size,
+  stringToHex,
+  type TypedDataDefinition,
+} from 'viem';
+import {
+  entryPoint06Abi,
+  entryPoint06Address,
+  getUserOperationHash,
+  type SmartAccount,
+  type SmartAccountImplementation,
+  toSmartAccount,
+  type UserOperation,
+  type WebAuthnAccount,
+} from 'viem/account-abstraction';
+
+export type CreateSmartAccountParameters = {
+  address: Address;
+  client: Client;
+  factoryData: Hex | undefined;
+  accountIndex: number;
+  account: OneOf<LocalAccount | WebAuthnAccount>;
+};
+
+export type CreateSmartAccountReturnType = Prettify<
+  SmartAccount<CoinbaseSmartAccountImplementation>
+>;
+
+export type CoinbaseSmartAccountImplementation = Assign<
+  SmartAccountImplementation<
+    typeof entryPoint06Abi,
+    '0.6',
+    { abi: typeof abi; factory: { abi: typeof factoryAbi; address: Address } }
+  >,
+  {
+    decodeCalls: NonNullable<SmartAccountImplementation['decodeCalls']>;
+    sign: NonNullable<SmartAccountImplementation['sign']>;
+  }
+>;
+
+/**
+ * @description Create a Coinbase Smart Account.
+ *
+ * @param parameters - {@link CreateSmartAccountParameters}
+ * @returns Coinbase Smart Account. {@link CreateSmartAccountReturnType}
+ *
+ * @example
+ * import { createSmartAccount } from 'viem/account-abstraction'
```

Comment:
What is the point of referencing this viem function here?  (need more context)

Reply:
Copy pasta from viems docs. Will remove